File: Python/TreeSort.py
class TreeNode:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

def insert(root, value):
    if root is None:
        return TreeNode(value)
    if value < root.value:
        root.left = insert(root.left, value)
    else:
        root.right = insert(root.right, value)
    return root

def in_order_traversal(root, result):
    if root:
        in_order_traversal(root.left, result)
        result.append(root.value)
        in_order_traversal(root.right, result)

def tree_sort(arr):
    root = None
   
    for value in arr:
        root = insert(root, value)

    arr.clear()  
    result = []
    in_order_traversal(root, result)
    arr.extend(result)

try:
    size = int(input("Enter the size of the list: "))
    arr = []
    for _ in range(size):
        while True:
            try:
                element = int(input("Enter a number: "))
                arr.append(element)
                break
            except ValueError:
                print("Invalid input. Please enter a valid number.")
    print("INPUT LIST:", arr)
    tree_sort(arr)
    print("SORTED LIST:", arr)
except KeyboardInterrupt:
    print("Program interrupted by the user.")
except Exception as e:
    print("INPUT MUST BE A NUMBER")

File: Javascript/TreeSort.js
class TreeNode {
    constructor(value) {
        this.value = value;
        this.left = null;
        this.right = null;
    }
}

function insert(root, value) {
    if (root === null) {
        return new TreeNode(value);
    }

    if (value < root.value) {
        root.left = insert(root.left, value);
    } else if (value > root.value) {
        root.right = insert(root.right, value);
    }

    return root;
}

function inOrderTraversal(root, result) {
    if (root !== null) {
        inOrderTraversal(root.left, result);
        result.push(root.value);
        inOrderTraversal(root.right, result);
    }
}

function treeSort(arr) {
    let root = null;

    for (let i = 0; i < arr.length; i++) {
        root = insert(root, arr[i]);
    }

    let sortedResult = [];
    inOrderTraversal(root, sortedResult);

    return sortedResult;
}

let arr = [5, 4, 3, 2, 1];
console.log("ARRAY BEFORE SORTING:");
console.log(arr);
const sortedArr = treeSort(arr);
console.log("ARRAY AFTER SORTING:");
console.log(sortedArr);
